src/rtnl.rs

```diff
@@ -31,8 +31,10 @@ impl<T, P> Nl for Vec<Rtattr<T, P>> where T: RtaType, P: Nl {
         )?;
         let mut vec = Vec::new();
         while size_hint > 0 {
-            let attr = Rtattr::deserialize(buf)?;
-            size_hint -= attr.asize();
+            let attr : Rtattr<T,P> = Rtattr::deserialize(buf)?;
+            size_hint = size_hint.checked_sub(attr.asize()).ok_or_else(|| {
```

Comment:
`.checked_sub()` is a good one to know about. There are some other places I should use this in `neli`.